metadata-io/src/main/java/com/linkedin/metadata/changeprocessor/ChangeProcessor.java

```diff
@@ -0,0 +1,18 @@
+package com.linkedin.metadata.changeprocessor;
```

Comment:
Split into -> SynchronizedChangePreProcessor & SynchronizedChangePostProcessor. We can discuss naming and impl here.

Reply:
I would suggest that updating ChangeProcessor to expose a ChangeProcessorType is a more flexible approach to typing the processors as opposed to strong typing by class.
What are your thoughts.

Reply:
I think that works, the only limitation that has is if the processor contract diverges based on entrypoint, but I think we should try to avoid that if possible anyway.